domain/crossmodelrelation/state/model/remoteapplication.go

```diff
@@ -548,7 +591,9 @@ INSERT INTO application_remote_offerer_status (*) VALUES ($remoteApplicationStat
 		Message:               sts.Message,
 		Data:                  sts.Data,
 		UpdatedAt:             sts.Since,
-	}).Run(); err != nil {
+	}).Run(); internaldatabase.IsErrConstraintForeignKey(err) {
+		return crossmodelrelationerrors.RemoteApplicationNotFound
```

Comment:
Is this true? There's 2 constraints on the table, so which one failed? It could be we're trying to insert a bad status value. I really dislike this "programme by exception" approach fwiw
Can we just do an explicit check.